packages/lib/response/service.ts

```diff
@@ -246,75 +235,131 @@ export const getResponse = async (responseId: string): Promise<TResponse | null>
-      skip: page ? ITEMS_PER_PAGE * (page - 1) : undefined,
-    });
-
-    const transformedResponses: TResponse[] = responses.map((responsePrisma) => ({
-      ...responsePrisma,
-      person: responsePrisma.person ? transformPrismaPerson(responsePrisma.person) : null,
-      tags: responsePrisma.tags.map((tagPrisma: { tag: TTag }) => tagPrisma.tag),
-    }));
-
-    return transformedResponses;
-  } catch (error) {
-    if (error instanceof Prisma.PrismaClientKnownRequestError) {
-      throw new DatabaseError("Database operation failed");
+      revalidate: SERVICES_REVALIDATION_INTERVAL,
     }
-
-    throw error;
-  }
-};
-
-export const getEnvironmentResponses = async (environmentId: string, page?: number): Promise<TResponse[]> => {
-  validateInputs([environmentId, ZId], [page, ZOptionalNumber]);
-
-  try {
-    const responses = await prisma.response.findMany({
-      where: {
-        survey: {
-          environmentId,
-        },
-      },
-      select: responseSelection,
-      orderBy: [
-        {
-          createdAt: "desc",
-        },
-      ],
-      take: page ? ITEMS_PER_PAGE : undefined,
-      skip: page ? ITEMS_PER_PAGE * (page - 1) : undefined,
-    });
-
-    const transformedResponses: TResponse[] = responses.map((responsePrisma) => ({
-      ...responsePrisma,
-      person: responsePrisma.person ? transformPrismaPerson(responsePrisma.person) : null,
-      tags: responsePrisma.tags.map((tagPrisma: { tag: TTag }) => tagPrisma.tag),
-    }));
-
-    return transformedResponses;
-  } catch (error) {
-    if (error instanceof Prisma.PrismaClientKnownRequestError) {
-      throw new DatabaseError("Database operation failed");
+  )();
+
+export const getEnvironmentResponses = async (environmentId: string, page?: number): Promise<TResponse[]> =>
+  unstable_cache(
+    async () => {
+      validateInputs([environmentId, ZId], [page, ZOptionalNumber]);
+
+      try {
+        const responses = await prisma.response.findMany({
+          where: {
+            survey: {
+              environmentId,
+            },
+          },
+          select: responseSelection,
+          orderBy: [
+            {
+              createdAt: "desc",
+            },
+          ],
+          take: page ? ITEMS_PER_PAGE : undefined,
+          skip: page ? ITEMS_PER_PAGE * (page - 1) : undefined,
+        });
+
+        const transformedResponses: TResponse[] = responses.map((responsePrisma) => ({
+          ...responsePrisma,
+          person: responsePrisma.person ? transformPrismaPerson(responsePrisma.person) : null,
+          tags: responsePrisma.tags.map((tagPrisma: { tag: TTag }) => tagPrisma.tag),
+        }));
+
+        return transformedResponses;
+      } catch (error) {
+        if (error instanceof Prisma.PrismaClientKnownRequestError) {
+          throw new DatabaseError("Database operation failed");
+        }
+
+        throw error;
+      }
+    },
+    [`getEnvironmentResponses-${environmentId}`],
```

Comment:
and the cache tag `getResponses-${environmentId}-${page}`

Reply:
do you mean tag or key?

also would it rather be `getResponsesByEnvironmentId-${environmentId}-${page}`?

Reply:
key, sorry ;-)
For the other functions you went for `getResponses-...args`,  but I think I would like `getResponsesByEnvironmentId-...args` better. It's easier to follow :-)

Reply:
Agree 👍🏾